api/instance.go

```diff
@@ -45,16 +55,15 @@ type InstanceCreate struct {
 	CreatorID int
 
 	// Related fields
-	EnvironmentID int `jsonapi:"attr,environmentId"`
+	EnvironmentID  int                 `jsonapi:"attr,environmentId"`
+	DataSourceList []*DataSourceCreate `jsonapi:"relation,dataSourceList"`
```

Comment:
We can keep this as is. For now, we only allow to specify an admin username/password in the instance creation request.